### PR TITLE
patch: implement support for api config in middleware input schemas

### PR DESCRIPTION
**Pull request by stevefrench39**
Adds `api` configuration support to the middleware input schemas, allowing consumers to pass API-level options through the standard middleware config surface. Updates all Cloudflare and Vercel runners/adapters plus the corresponding tests and skill documentation.

### CODE REVIEW

File: src/utils/route-matching.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  pattern: string,
+): boolean => {
+  const normalized = normalizePathPattern(pattern)
+
+  if (normalized === requestPath) {
+    return true
+  }
+
+  return requestPath.startsWith(`${normalized}/`)
+}
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/utils/route-matching.ts:25]: ✅ Fixed. `matchesSegmentBoundaryPath` now treats a normalized pattern of `/` as matching all request paths, and a test was added for root patterns.

File: src/schemas/middleware-options.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  lockPageSlug: LockPageSlugSchema.optional(),
+  cspMode: CSPModeSchema.optional(),
+  cspDirectives: NonceFreeCSPDirectivesSchema.optional(),
+})
+
+export const WebsiteMiddlewareConfigNextJsSchema = z.object({
+  lockPageSlug: LockPageSlugSchema.optional(),
+  cspMode: CSPModeSchema.optional(),
+  cspDirectives: NextJsNonceFreeCSPDirectivesSchema.optional(),
+})
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/schemas/middleware-options.ts:114]: ✅ Fixed. Removed `.optional()` from `lockPageSlug` in all website schemas so the default `/maintenance` is applied when callers provide `website: {}`.

File: src/schemas/middleware-options.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  {
+    message: "Path pattern must be a valid absolute path starting with /",
+  },
+)
+
+export const PathPatternsSchema = z.array(PathPatternSchema)
+
+export const LockPageSlugSchema = ValidLockPageSlugSchema.default(
+  "/maintenance",
+).transform((val) => (val.startsWith("/") ? val : `/${val}`))
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/schemas/middleware-options.ts]: ✅ Fixed. Updated `LockPageSlugSchema` transform so an empty string falls back to `/maintenance` instead of becoming `/`.

File: src/schemas/middleware-options.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+export const ApiResponseSchema = z.object({
+  status: z.number().default(DEFAULT_API_LOCK_STATUS),
+  body: z.string().default(DEFAULT_API_LOCK_BODY),
+  headers: z.array(ApiResponseHeaderSchema).optional(),
+})
+
+export const ApiMiddlewareConfigSchema = z.object({
+  basePaths: PathPatternsSchema,
+  response: ApiResponseSchema.default({
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/schemas/middleware-options.ts:132]: ✅ Fixed. Added `.min(1)` to `ApiMiddlewareConfigSchema.basePaths` so empty arrays are rejected.

File: src/utils/vercel/delete-edge-value.test.ts [hunk 5]
```diff
@@ -60,8 +60,7 @@ describe("deleteEdgeValue", () => {
       vercelApiToken: "test-token",
       appwardenApiToken: "test-token",
       requestUrl: new URL("https://example.com"),
-      lockPageSlug: "/locked",
-    } as VercelProviderContext
+    } as unknown as VercelProviderContext
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/utils/vercel/delete-edge-value.test.ts]: ✅ Fixed. Replaced `as unknown as VercelProviderContext` with a typed `createMockContext` helper that fills in minimal required stubs.

File: src/utils/vercel/get-lock-value.test.ts [hunk 6]
```diff
@@ -81,8 +81,7 @@ describe("getLockValue", () => {
       vercelApiToken: "test-token",
       appwardenApiToken: "test-token",
       requestUrl: new URL("https://example.com"),
-      lockPageSlug: "/locked",
-    } as VercelProviderContext
+    } as unknown as VercelProviderContext
```
[the remark this answers is not in the document]
Comment (stevefrench39) [reply src/utils/vercel/get-lock-value.test.ts]: ✅ Fixed. Replaced `as unknown as VercelProviderContext` with typed `Pick<VercelProviderContext, 'cacheUrl' | 'keyName' | 'provider'>` contexts.
